wooey/models/favorite.py

```diff
@@ -0,0 +1,18 @@
+from django.db import models
+from django.conf import settings
+from ..django_compat import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+
+class Favorite(models.Model):
+
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, related_name='favorites', blank=True, null=True)
+
+    content_type = models.ForeignKey(ContentType)
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey('content_type', 'object_id')
```

Comment:
Which things are going to be favorited? I only ask because I sort of hate GFKs.

Consider this: You are fetching a user's favorites. There are 100 things. Now to get the thing that is favorited, you have an additional 2 queries per favorite (content table lookup, then model lookup). You can avoid this by fetching all favorites/gfks/etc. and making a huge dictionary for lookups -- but that can take up a ton of memory and cause the server to crash.

Do you know a good way around this? Maybe a custom manager can handle this because prefetches as far as I remember break down with GFKs as well.